runtime/compiler/control/JITaaSCompilationThread.cpp

```diff
@@ -2415,20 +2431,35 @@ ClientSessionData::~ClientSessionData()
    _romMapMonitor->destroy();
    _systemClassMapMonitor->destroy();
    _sequencingMonitor->destroy();
+   _unloadedClassAddresses.destroy();
    jitPersistentFree(_vmInfo);
    }
 
 void
-ClientSessionData::processUnloadedClasses(const std::vector<TR_OpaqueClassBlock*> &classes)
+ClientSessionData::processUnloadedClasses(JITaaS::J9ServerStream *stream, const std::vector<TR_OpaqueClassBlock*> &classes)
    {
    if (TR::Options::getVerboseOption(TR_VerboseJITaaS))
       TR_VerboseLog::writeLineLocked(TR_Vlog_JITaaS, "Server will process a list of %u unloaded classes for clientUID %llu", (unsigned)classes.size(), (unsigned long long)_clientUID);
+
+   if (_requestUnloadedClasses)
+      {
+      stream->write(JITaaS::J9ServerMessageType::getUnloadedClassRanges, JITaaS::Void());
+      auto unloadedClassRanges = std::get<0>(stream->read<std::vector<TR_AddressRange> >());
+      OMR::CriticalSection getUnloadedClasses(getROMMapMonitor());
+      _unloadedClassAddresses.setRanges(unloadedClassRanges);
+      _requestUnloadedClasses = false;
+      }
+
    OMR::CriticalSection processUnloadedClasses(getROMMapMonitor());
-   for (TR_OpaqueClassBlock *clazz : classes)
+
+   for (auto clazz : classes)
       {
+      _unloadedClassAddresses.add((uintptrj_t)clazz);
```

Comment:
This line could be avoided when `_requestUnloadedClasses` was found true because we just initialized the most up-to-date set of ranges.